lib/manager/components/LabelEditor.js

```diff
@@ -0,0 +1,236 @@
+// @flow
+
+import React from 'react'
+import {
+  Button,
+  Col,
+  ControlLabel,
+  FormControl,
+  FormGroup,
+  HelpBlock,
+  ListGroup,
+  ListGroupItem,
+  Panel,
+  Row
+} from 'react-bootstrap'
+import update from 'react-addons-update'
+import { connect } from 'react-redux'
+
+import { createLabel, updateLabel } from '../../manager/actions/labels'
+import { validationState } from '../util'
+import type { ManagerUserState } from '../../types/reducers'
+import type { Label } from '../../types'
+
+type Props = {
+  createLabel: Function,
+  label: Label,
+  onDone?: Function,
+  projectId: string,
+  updateLabel: Function,
+  user: ManagerUserState
+}
+
+type State = {
+  labelIsNew?: boolean,
+  newLabel: Label,
+  validation: {
+    description: boolean,
+    name: boolean
+  }
+}
+
+class LabelEditor extends React.Component<Props, State> {
+    state = {
+      validation: {
+        name: true,
+        description: true
+      },
+      newLabel: {...this.props.label},
+      labelIsNew: false
+    }
+
+    componentDidMount = () => {
+      // If we didn't get a label passed in, declare the label to be new
+      if (Object.keys(this.props.label).length === 0) {
+        this.setState({
+          labelIsNew: true,
+          // Default to a nice orange
+          // we can safely set a partial object here
+          // $FlowFixMe
```

Comment:
Or you can pass the default validation state.

Reply:
Much cleaner thank you! Refactored.